internal/ethapi/celo_block.go

```diff
@@ -0,0 +1,166 @@
+package ethapi
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	gasPriceMinimumABIJson = `[{"inputs":[{"internalType":"uint256","name":"gasPriceMinimum","type":"uint256"}],"name":"GasPriceMinimumUpdated","outputs":[],"type":"event"}]`
+	gasPriceMinimumABI     abi.ABI
+)
+
+func init() {
+	parsedAbi, _ := abi.JSON(strings.NewReader(gasPriceMinimumABIJson))
+	gasPriceMinimumABI = parsedAbi
+}
+
+// PopulatePreGingerbreadBlockFields populates the baseFee and gasLimit fields of the block for pre-gingerbread blocks
+func PopulatePreGingerbreadBlockFields(ctx context.Context, backend CeloBackend, block *types.Block) *types.Block {
+	newHeader := PopulatePreGingerbreadHeaderFields(ctx, backend, block.Header())
+	block = block.WithSeal(newHeader)
+	return block
+}
+
+// PopulatePreGingerbreadHeaderFields populates the baseFee and gasLimit fields of the header for pre-gingerbread blocks
+func PopulatePreGingerbreadHeaderFields(ctx context.Context, backend CeloBackend, header *types.Header) *types.Header {
+	isGingerbread := backend.ChainConfig().IsGingerbread(header.Number)
+	if isGingerbread {
+		return header
+	}
+
+	var (
+		baseFee  *big.Int
+		gasLimit *big.Int
+	)
+
+	fields, err := rawdb.ReadPreGingerbreadAdditionalFields(backend.ChainDb(), header.Hash())
```

Comment:
Shouldn't `err` be checked before accessing `fields`?

Reply:
Added logging for the error: a67a0adbcafd7b6fec8c9887b735a84c8b3cec70